scipy/stats/tests/test_stats.py

```diff
@@ -6732,47 +6732,52 @@ def test_pointbiserial():
     assert_equal(res.correlation, res.statistic)
 
 
-def test_obrientransform():
+@array_api_compatible
+def test_obrientransform(xp):
     # A couple tests calculated by hand.
-    x1 = np.array([0, 2, 4])
+    x1 = xp.asarray([0, 2, 4])
     t1 = stats.obrientransform(x1)
-    expected = [7, -2, 7]
-    assert_allclose(t1[0], expected)
+    expected = xp.asarray([7., -2., 7.])
+    xp_assert_close(t1[0, :], expected)
 
-    x2 = np.array([0, 3, 6, 9])
+    x2 = xp.asarray([0, 3, 6, 9])
     t2 = stats.obrientransform(x2)
-    expected = np.array([30, 0, 0, 30])
-    assert_allclose(t2[0], expected)
+    expected = xp.asarray([30., 0., 0., 30.])
+    xp_assert_close(t2[0, :], expected)
 
     # Test two arguments.
     a, b = stats.obrientransform(x1, x2)
-    assert_equal(a, t1[0])
-    assert_equal(b, t2[0])
+    xp_assert_close(a, t1[0, :])
+    xp_assert_close(b, t2[0, :])
 
     # Test three arguments.
     a, b, c = stats.obrientransform(x1, x2, x1)
-    assert_equal(a, t1[0])
-    assert_equal(b, t2[0])
-    assert_equal(c, t1[0])
+    xp_assert_close(a, t1[0, :])
+    xp_assert_close(b, t2[0, :])
+    xp_assert_close(c, t1[0, :])
 
     # This is a regression test to check np.var replacement.
     # The author of this test didn't separately verify the numbers.
-    x1 = np.arange(5)
-    result = np.array(
+    x1 = xp.arange(5)
+    result = xp.asarray(
       [[5.41666667, 1.04166667, -0.41666667, 1.04166667, 5.41666667],
        [21.66666667, 4.16666667, -1.66666667, 4.16666667, 21.66666667]])
-    assert_array_almost_equal(stats.obrientransform(x1, 2*x1), result, decimal=8)
+    xp_assert_close(stats.obrientransform(x1, 2*x1), result)
 
     # Example from "O'Brien Test for Homogeneity of Variance"
     # by Herve Abdi.
     values = range(5, 11)
-    reps = np.array([5, 11, 9, 3, 2, 2])
-    data = np.repeat(values, reps)
-    transformed_values = np.array([3.1828, 0.5591, 0.0344,
-                                   1.6086, 5.2817, 11.0538])
-    expected = np.repeat(transformed_values, reps)
+    reps = xp.asarray([5, 11, 9, 3, 2, 2])
```

Comment:
Can we keep using NumPy to compute the data and expected values and convert them to `xp`-type only when needed?